### ISSUE TITLE
Uptime Integration with Infrastructure UI / Log UI

### ISSUE THREAD

**Issue by dov0211** [elastic/uptime#6]
Capturing the notes from the mail [thread](https://groups.google.com/a/elastic.co/forum/?utm_medium=email&utm_source=footer#!msg/uptime/J6W7ko-Qxr0/q3OfHiS4GQAJ). 
Integrations between observability solutions (Uptime + Infrastructure IU/ Log UI) will be valuable for our existing customers and appears to be a low hanging fruit 
Heartbeat reports the `monitor.ip` (the monitored instance), in case we have a running Filebeat or Metricbeat on that same instance, we can provide a link to Infrastructure or log UI 

_[Andrew C]_
We currently use `host.*` for the beat itself (as in host.geo)
ECS doesn't fit us *that* well, we could:

1. Switch all `host` fields to `observer.*`
2. Use a new `target` namespace for what we're monitoring.

The rationale is that host doesn't fit that well given the ECS definition of ' the host on which the event happened'.

_[Nicolas R]_
host.ip sounds like the right place for me and moving the host data to observer also the right move.

**Comment by legrego** [comment]
Something to keep in mind as this is getting developed: The introduction of [Feature Controls](https://github.com/elastic/kibana/pull/31652) will make it trivial for kibana admins to disable apps for groups of users and/or spaces. We should design this integration with the assumption that Infra/Logs will not always be available to the end user.

We'll be providing more guidance soon on how to perform these checks -- don't worry, we've made it easy!

**Comment by dov0211** [comment]
@legrego as this integration been developed by @justinkambic, can you provide us with some guidance?

**Comment by justinkambic** [comment]
++ this is something we'll want to integrate into the features I am actively adding to our UI.

**Comment by legrego** [comment]
@dov0211 @justinkambic absolutely!

For your immediate needs, you can figure out if Infra/Logs are available for the current user/space by doing something like this on the client-side:

```ts
import { uiCapabilities } from 'ui/capabilities';

const isInfraAppAvailable = uiCapabilities.infrastructure && uiCapabilities.infrastructure.show;
const isLogsAppAvailable = uiCapabilities.logs && uiCapabilities.logs.show;
```

If you're in a React app, we also have UICapabilities available via context, so you don't have to use the static import if you don't want to. See [`ui_capabilities_provider`](https://github.com/elastic/kibana/blob/master/src/legacy/ui/public/capabilities/react/ui_capabilities_provider.tsx) and [`inject_ui_capabilities`](https://github.com/elastic/kibana/blob/master/src/legacy/ui/public/capabilities/react/inject_ui_capabilities.tsx). The Infra/Logs apps are using these react constructs, so you can check their plugin out for an example if you want to take this route.

The two "show" capabilities are registered in the Infra plugin via its c[all to `xpackMainPlugin.registerFeature()`](https://github.com/elastic/kibana/blob/master/x-pack/plugins/infra/server/kibana.index.ts#L25-L83). Uptime has a call to this function as well, which is what allows it to appear in Spaces Management and Role Management as a feature we can enable/disable.

I'm happy to go into more detail in a zoom if that's helpful, but this should get you started in the meantime. Let me know if there's anything else I can do to help!

**Comment by dov0211** [comment]
thank you @legrego , I guess this also implies to APM (in addition to logs and Infra)

**Comment by legrego** [comment]
@dov0211 yes, for APM, you can check `uiCapabilities.apm.show`

**Comment by andrewvc** [comment]
Was completed and shipped in 7.3
